src/features/views/components/ViewColumnDialog/SurveyResponseConfig.tsx

```diff
@@ -0,0 +1,226 @@
+                onChange={(evt, value) => {
+                  if (value !== null) {
+                    setSelectedQuestion(value);
+                    const columns = makeColumnsText(value);
+                    if (columns !== undefined) {
+                      onOutputConfigured(columns);
+                    }
+                  }
+                }}
+                options={questionFromSurvey || []}
+                renderInput={(params) => (
+                  <TextField
+                    {...params}
+                    inputProps={{
+                      ...params.inputProps,
+                    }}
+                    label={intl.formatMessage({
+                      id: 'misc.views.columnDialog.choices.surveyResponse.questionField',
+                    })}
+                    variant="standard"
+                  ></TextField>
+                )}
+                value={selectedQuestion}
+              />
+            ) : (
+              ''
+            )}
+            {surveyId &&
+            selectedQuestion &&
+            selectedQuestion?.question.response_type === RESPONSE_TYPE.TEXT ? (
+              <Typography sx={{ marginTop: '15' }} variant="h6">
+                <Msg id="misc.views.columnDialog.choices.surveyResponse.textField" />
+              </Typography>
+            ) : selectedQuestion?.question.response_type ===
+              RESPONSE_TYPE.OPTIONS ? (
+              <TextField
+                label={intl.formatMessage({
+                  id: 'misc.views.columnDialog.choices.surveyResponse.optionsLabel',
+                })}
+                onChange={(evt) => {
+                  const columns = makeColumnsOption(evt.target.value);
+                  if (columns !== undefined) {
+                    onOutputConfigured(columns);
+                  }
+                }}
+                select
+                variant="standard"
+              >
+                <ListSubheader>
+                  <FormattedMessage id="misc.views.columnDialog.choices.surveyResponse.allOptionsHeader" />
+                </ListSubheader>
+                <MenuItem value={SURVEY_QUESTION_OPTIONS.ALL_OPTIONS}>
+                  <FormattedMessage id="misc.views.columnDialog.choices.surveyResponse.allOptions" />
+                </MenuItem>
+                <MenuItem value={SURVEY_QUESTION_OPTIONS.ALL_OPTIONS_SEPARATED}>
+                  <FormattedMessage id="misc.views.columnDialog.choices.surveyResponse.allOptionsSeparated" />
+                </MenuItem>
+                <ListSubheader>
+                  <FormattedMessage id="misc.views.columnDialog.choices.surveyResponse.oneOption" />
+                </ListSubheader>
+                {selectedQuestion?.question.options?.map((option) => (
+                  <MenuItem key={option.id} value={option.id}>
+                    {option.text}
+                  </MenuItem>
+                ))}
+              </TextField>
+            ) : (
+              ''
+            )}
```

Comment:
I am curious about these empty strings in the ternaries - do they do something different from the conditional render using "&&" ?